relay-server/src/actors/project.rs

```diff
@@ -781,8 +781,12 @@ impl Project {
             return;
         }
 
-        self.state_channel = None;
-        self.state = Some(state.clone());
+        match self.expiry_state() {
+            // If the new state is invalid but the old one still usable, keep the old one.
+            ExpiryState::Updated(old) | ExpiryState::Stale(old) if state.invalid() => state = old,
+            // If the new state is valid or the old one is expired, always use the new one.
+            _ => self.state = Some(state.clone()),
```

Comment:
Is there a reason to fall back to the old state, if both are expired? Do we assume that the new state is more up-to-date even if both are stale?